pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -290,6 +298,69 @@ func (c *Cluster) startMons(targetCount int) error {
 	return nil
 }
 
+func (c *Cluster) configureStretchCluster(mons []*monConfig) error {
+	if !c.spec.IsStretchCluster() {
+		return nil
+	}
+
+	if err := c.assignStretchMonsToZones(mons); err != nil {
+		return errors.Wrap(err, "failed to assign mons to zones")
+	}
+
+	// Enable the mon connectivity strategy
+	if err := client.EnableStretchElectionStrategy(c.context, c.ClusterInfo); err != nil {
+		return errors.Wrap(err, "failed to enable stretch cluster")
+	}
+
+	// Create the default crush rule for stretch clusters, that by default will also apply to all pools
+	if err := client.CreateDefaultStretchCrushRule(c.context, c.ClusterInfo, &c.spec, c.stretchFailureDomainName(), c.spec.Mon.StretchCluster.SubFailureDomain); err != nil {
+		return errors.Wrap(err, "failed to create default stretch rule")
+	}
+
+	return nil
+}
+
+func (c *Cluster) assignStretchMonsToZones(mons []*monConfig) error {
+	var arbiterZone string
+	for _, zone := range c.spec.Mon.StretchCluster.Zones {
+		if zone.Arbiter {
+			arbiterZone = zone.Name
+			break
+		}
+	}
+	if arbiterZone == "" {
+		return errors.Errorf("failed to find the arbiter zone")
+	}
+
+	// Set the location for each mon
+	domainName := c.stretchFailureDomainName()
+	for _, mon := range mons {
+		if mon.Zone == arbiterZone {
+			// remember the arbiter mon to be set later in the reconcile after the OSDs are configured
+			c.arbiterMon = mon.DaemonName
+		}
+		logger.Infof("setting mon %q to stretch %s=%s", mon.DaemonName, domainName, mon.Zone)
+		if err := client.SetMonStretchZone(c.context, c.ClusterInfo, mon.DaemonName, domainName, mon.Zone); err != nil {
+			return errors.Wrapf(err, "failed to set mon %q zone", mon.DaemonName)
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) ConfigureArbiterIfStretchCluster() error {
+	if c.arbiterMon == "" {
```

Comment:
Is it really necessary? Will this be triggered if not stretch cluster?

Reply:
Missed? `c.arbiterMon` content must have been validated earlier, right?

Reply:
And if we remove the check, do we need this method at all? I believe we can just call ` client.SetMonStretchTiebreaker()` directly wherever `ConfigureArbiter()` is called.

Reply:
Sure we can just call it directly

Reply:
`ConfigureArbiter()` is called from two different places so I'll actually keep the helper to keep the args simpler in one place.